ably/realtime/websockettransport.py

```diff
@@ -72,6 +77,22 @@ async def ws_connect(self, ws_url, headers):
         except (WebSocketException, socket.gaierror) as e:
             raise AblyException(f'Error opening websocket connection: {e}', 400, 40000)
 
+    async def on_protocol_message(self, msg):
+        self.on_activity()
+        log.info(f'WebSocketTransport.on_protocol_message(): receieved protocol message: {msg}')
+        if msg['action'] == ProtocolMessageAction.CONNECTED:
+            connection_details = msg.get('connectionDetails')
```

Comment:
Yeah that's not a bad idea but i'm not sure whether there's a sensible way to do that, connection details are parsed to the dataclass in `ConnectionManager` but the implementation for `max_idle_interval` lives in the transport. I think it might be a good idea to move protocol message handling to the transport but without doing that I'm not sure what else to do.

Reply:
Connection Manager is accessible from transport, maybe importing it might work?

Reply:
Yeah true, but that would have caused a circular import. In the end I went for moving all protocol message handling into the `WebSocketTransport` class ([a22330d](https://github.com/ably/ably-python/pull/405/commits/a22330d5fe9dae4a1e31a740f7f788f8945b8387)). This means that the `ConnectionDetails` class is instantiated while the `WebSocketTransport` still has the message in scope so now it works. Also it makes the `ConnectionManager` class a lot simpler which is a bonus 🙂